src/descriptor.rs

```diff
@@ -274,13 +277,21 @@ impl Descriptor {
     ///
     /// ```
     /// # use bitcoin_bosd::{Descriptor, DescriptorType, descriptor::P2TR_LEN};
-    /// let payload = [0u8; P2TR_LEN]; // all zeros, don't use in production
+    /// let payload = [2u8; P2TR_LEN]; // valid X-only public key, but don't use in production
     /// let desc = Descriptor::new_p2tr(&payload);
     /// # assert_eq!(desc.type_tag(), DescriptorType::P2tr);
-    /// # assert_eq!(desc.payload(), [0u8; P2TR_LEN]);
+    /// # assert_eq!(desc.payload(), [2u8; P2TR_LEN]);
     /// ```
     pub fn new_p2tr(payload: &[u8; P2TR_LEN]) -> Self {
         let type_tag = DescriptorType::P2tr;
+
+        // If using `rust-bitcoin` then we can validate internally the X-only public key.
+        #[cfg(feature = "address")]
+        assert!(
+            XOnlyPublicKey::from_slice(payload).is_ok(),
+            "payload is not a valid X-only public key!"
+        );
```

Comment:
I really dislike panics in production code, especially since the documentation doesn't make it clear that an invalid key will result in a panic. Would advise either having this return `Option<Self>`, or (if absolutely needed) carefully documenting the panic.

Reply:
What if we use a `debug_assert!`?

Reply:
Also this can lead to funds being lost, maybe that should be outweighed against the panic. I honestly don't know how to proceed here.
This is a good check. Should be done, even in production code. Maybe we can change the function signature to return a `Result` and in the docstring warn the user that checks are only performed if the crate has `address` feature enabled?

Reply:
This feature gating changes the behavior of the function to the extent that I wonder if it's better to keep the existing "unchecked" functionality, but add some kind of `new_p2tr_checked` that's gated and returns `Option<Self>`. It just seems like having a constructor that panics on bad input in a feature-gated way is a big footgun.

Reply:
I like the `new_p2tr_checked` and returning an `Option` however should strive for consistency with Rust and be `checked_new_p2tr`, what you think?

Reply:
Oh hmm, I'm not sure what's more idiomatic in this particular case. Maybe someone like @delbonis can chime in? But overall it sounds like we're in agreement: that it's much safer to have the functionality be consistent regardless of feature gating.